### PR TITLE
[String] Added test for replaceMatches

### LINKED ISSUE #33905: [String] Add tests for replaceMatches()
**Symfony version(s) affected**: 5.0

**Description**  
I can't find tests (or a method description) for the method `String:replaceMatches()`. Its behavior is more complex than it looks at first sight, so it's important to do this.

### PR DESCRIPTION
**Pull request by honarkhah**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #33905
| License       | MIT
| Doc PR        | 

Added test for String:replaceMatches

### CODE REVIEW

**Comment by honarkhah** [comment]
https://travis-ci.org/symfony/symfony/builds/595610977

File: src/Symfony/Component/String/Tests/AbstractUnicodeTestCase.php [hunk 1]
```diff
@@ -405,6 +405,16 @@ public static function provideReplace(): array
         );
     }
 
+    public static function provideReplaceMatches(): array
+    {
+        return array_merge(
+            parent::provideReplaceMatches(),
+            [
+                ['This is a dj-vu situation!', 'This is a déjà-vu situation!', '/([à-ú])/', ''],
```
Comment (OskarStark) [inline src/Symfony/Component/String/Tests/AbstractUnicodeTestCase.php]: Nice 😋😁

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/String/Tests/AbstractUnicodeTestCase.php (hunk 1 above)
Comment (honarkhah) [reply src/Symfony/Component/String/Tests/AbstractUnicodeTestCase.php]: what is the issue with continuous-integration/appveyor?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @honarkhah.

**Pull request merged**